realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -138,6 +143,279 @@ public boolean load() {
+
+        collection.setBinary(getColumnIndex(fieldName), value);
+    }
+
+    /**
+     * Sets the {@code Date} value of the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or field isn't date.
+     */
+    public void setDate(@Nonnull String fieldName, @Nullable Date value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        collection.setDate(getColumnIndex(fieldName), value.getTime());
+    }
+
+    /**
+     * Sets a reference to another object on the given field in all of the objects in the result collection.
+     *
+     * @param fieldName name of the field to update.
+     * @param value new value for the field.
+     * @throws IllegalArgumentException if field name doesn't exist or is of the wrong type.
+     */
+    public void setObject(@Nonnull String fieldName, @Nullable RealmObject value) {
+        verifyString(fieldName);
+
+        realm.checkIfValid();
+
+        if (checkAndSetNullValue(fieldName, value)) { return; }
+
+        long columnIndex = getColumnIndex(fieldName);
+
+        if (!(value instanceof RealmObjectProxy)) {
+            throw new IllegalArgumentException("A field value must be a managed Realm object.");
+        }
+
+        ProxyState proxyState = ((RealmObjectProxy) value).realmGet$proxyState();
+        if (proxyState.getRealm$realm() != realm) {
+            throw new IllegalArgumentException("A field value must belong to the same Realm as its owner.");
+        }
+
+        Row row = proxyState.getRow$realm();
+        if (row == null) {
+            throw new IllegalArgumentException("A field value must be a managed Realm object.");
+        }
+        if (!(row instanceof UncheckedRow)) {
+            throw new IllegalArgumentException("WTF?");
+        }
+
+        collection.setObject(columnIndex, (UncheckedRow) row);
+    }
+
+    /**
+     * Sets the value for the given field  in all of the objects in the result collection, converting {@code String}
+     * representations of numbers and booleans to their appropriate type. For example {@code "10"}
+     * will be converted to {@code 10} if the field type is {@code int}.
+     * <p>
+     * Using the typed setters is faster than using this method.
+     *
+     * @throws IllegalArgumentException if field name doesn't exist or if the input value cannot be converted
+     * to the appropriate input type.
+     * @throws NumberFormatException if a String based number cannot be converted properly.
+     * @throws RealmException if the field is a {@link io.realm.annotations.PrimaryKey} field.
+     */
+    @SuppressWarnings("unchecked")
+    public void set(@Nonnull String fieldName, @Nullable String strValue) {
```

Comment:
The use case for `DynamicRealmObject.set()` was CSV, not JSON, since JSON is more about whole objects while CSV is about dynamically parsing single lines and fields.

I doubt that would be the case for `RealmResults.setX()`. I see that more used in cases like "Mark all emails as read"-scenarios, aka. UI driven events that operate on the local Realm.

So IMO I would remove it unless you can see a use case?